src/historywork/GetHistoryArchiveStateWork.cpp

```diff
@@ -59,6 +66,23 @@ GetHistoryArchiveStateWork::doWork()
                 return State::WORK_FAILURE;
             }
         }
+        else if (state == State::WORK_FAILURE && archive)
+        {
+            if (isWellKnown(mSeq))
+            {
+                // Archive is corrupt if it's missing a well-known file
+                CLOG(ERROR, "History") << fmt::format(
```

Comment:
NB: this changes what https://github.com/stellar/stellar-core/pull/2784 should do (as logging is already done here)